searchlib/src/vespa/searchlib/tensor/hnsw_graph.h

```diff
@@ -63,9 +63,25 @@ struct HnswGraph {
     
     void set_link_array(uint32_t docid, uint32_t level, const LinkArrayRef& new_links);
 
-    void set_entry_node(uint32_t docid, int32_t level) {
-        entry_docid = docid;
-        entry_level = level;
+    struct EntryNode {
+        uint32_t docid;
+        int32_t level;
+        EntryNode() : docid(0), level(-1) {}
```

Comment:
Consider adding a constructor that takes docid and level, so we can keep that one liner `set_entry_node(EntryNode(my_docid, my_level))`, instead of the four lines it has changed into now in various places.